uber/site_sections/preregistration.py

```diff
@@ -661,13 +661,41 @@ def not_found(self, id, message=''):
 
     def abandon_badge(self, session, id):
         attendee = session.attendee(id)
-        success_message = "Sorry you can't make it! We hope to see you next year!"
-        failure_message = "You cannot abandon your badge because you are the leader of a group."
+        if attendee.amount_paid:
+            failure_message = "Something went wrong with your refund. Please contact us at {}."\
+                .format(c.REGDESK_EMAIL)
+        else:
+            success_message = "Sorry you can't make it! We hope to see you next year!"
+            if attendee.is_group_leader:
+                failure_message = "You cannot abandon your badge because you are the leader of a group."
+            else:
+                failure_message = "You cannot abandon your badge for some reason. Please contact us at {}."\
+                    .format(c.REGDESK_EMAIL)
 
-        # a group leader may not abandon their badge
-        if attendee.is_group_leader:
+        if not attendee.can_abandon_badge and not attendee.can_refund_badge:
             raise HTTPRedirect('confirm?id={}&message={}', id, failure_message)
 
+        if attendee.amount_paid:
+            amount_refunded = 0
+            if not all(txn.stripe_id
```

Comment:
This is a little strict, but basically we want to make REALLY sure that attendees can't cause horrible problems with self-service refunds, so we make sure that every transaction in their table has a stripe ID and is a payment (vs a refund) before we start processing any of them. If any don't fit the criteria, we tell them to contact us.